### DISCUSSION TITLE
Strange problem when using sys.path.append()

### DISCUSSION THREAD

**Discussion by kcvinker** [brentvollebregt/auto-py-to-exe#396] [General]
Hi all,
I wrote a gui library and wrote a little program with it. Everything works fine but I created an exe file and ran it. Then it showed this message box.
![image](https://user-images.githubusercontent.com/8840907/236353295-347f9d68-763c-44f9-a76b-4a7175ebc9c2.png)

My library depends upon ctypes. It says it cannot find ctypes.
In my script, this is how I connect my gui library.
```python
import sys
sys.path.append(r"C:\Users\kcvin\OneDrive\Programming\Python\PyForms")
```
So, I just commented out these two lines and copy pasted this script to "PyForms" directory. Then I re compiled my program as an exe and it worked.  
![image](https://user-images.githubusercontent.com/8840907/236353762-189281c9-b60f-4173-a866-92c8edc5d197.png)

So what's the problem with `sys.path.append`

**Comment by brentvollebregt** [comment]
Discussed in https://github.com/brentvollebregt/auto-py-to-exe/issues/395

**Reply by kcvinker** [reply]
I am sorry, but in #395, I just explained what's the problem is. But no one yet suggested a solution.

**Reply by brentvollebregt** [reply]
@kcvinker you had closed the issue as complete - were your intentions to move the issue to a discussion here? I had assumed you had fixed it as you said "Oh I see. Okay. Anyhow, I found the problem."

On your question, you said you removed those two lines **and** copy-pasted this script to "PyForms" directory. I have no context of what "this script" is but based on what you're saying, the issue doesn't sound like it lies in `sys.path.append`.

I understand you have said that these two lines are what you use to connect to your GUI library:

```python
import sys
sys.path.append(r"C:\Users\kcvin\OneDrive\Programming\Python\PyForms")
```

But that is simply extending the path.

**Reply by kcvinker** [reply]
@brentvollebregt 
Thank you for the reply. Yes, that's right. Those two lines are connecting my gui library. Beneath those lines, comes the import statements from the gui library. For example the `Form` class, which is a representation of a `Window`. this class imports the `ctypes` module. So when we run the exe file created by your program, it says it can't find the ctypes  module.  But it safely find all the dependencies, only if I comment out the above noted lines and place my script near to the gui library.

**Reply by brentvollebregt** [reply]
Are you able to provide a stack trace?

That `sys.path.append` is not connecting anything - it is adding a path where the Python interpreter searches for modules. This hints that you may just have a referencing issue - you may be trying to import something that Python can't see (which would be an environment / implementation issue).

**Reply by kcvinker** [reply]
@brentvollebregt 
Sorry, I am not able to provide a stack trace. Yes, `sys.path.append()` is adding a path where the Python interpreter searches for modules. And I can run my script successfully with it. But when I run the exe file, it says, it can't find the modules. That's the problem.
